docs/source/developer/installing.rst

```diff
@@ -121,6 +146,54 @@ with the arguments of your choosing, e.g.:
 
     (py37) $ pytest tests/test_s3.py -x -k replication
 
-Note you'll have to environment variables setup appropriately per the tox.ini
+Note you'll have to set up environment variables appropriately per the tox.ini
 for provider credentials.
 
+
+Installing in Your Own Virtual Environment
+------------------------------------------
+
+Running directly from a tox sandbox, while very easy to set up, might not be
+the most comfortable way of working. You might want to create your own virtual
+environment and use that for running Custodian. This can be done using the ``venv``
+module. It can be done right inside the cloned Cloud Custodian repository:
+
+.. code-block:: bash
+
+    $ python3 -m venv .
+
+The above command assumes the current directory is the Cloud Custodian checkout.
+
+Next, you'll need to install all the development dependencies and set up the
+cloud-custodian directory as a development mode package:
+
+.. code-block:: bash
+
+    $ source bin/activate
+    (cloud-custodian) $ pip install -e .
+    (cloud-custodian) $ pip install -r requirements-dev.txt
+
+Once this is done, ``pytest`` can be used to run the tests, as mentioned in the
+previous section:
+
+.. code-block:: bash
+
+    (cloud-custodian) $ pytest
+
+You will need to set the following environment variables to successfully run the
+full test suite:
```

Comment:
rather than documenting this, have people source `test.env` file which is accommodates updates rather than trying to capture a full list here.

Reply:
Right, that's a lot better. Will change.